avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1738,12 +1775,37 @@ private static boolean isInstanceOfAvroSchemaTypeForSimpleSchema(Schema fieldSch
     }
     for (Class type : classes) {
       if (type.isInstance(value)) {
-        return true;
+        if (fieldSchema.type() == Schema.Type.BYTES
+            && fieldSchema.parameters() != null
+            && fieldSchema.parameters().containsKey(CONNECT_AVRO_FIXED_SIZE)) {
+          if (fixedValueSizeOf(value)
+              == Integer.parseInt(fieldSchema.parameters().get(CONNECT_AVRO_FIXED_SIZE))) {
```

Comment:
This needs to change too, but it is trickier. AvroData accepts data in `byte[]` and `ByteBuffer` formats, in addition to `GenericFixed`. The latter can be matched by name, but the former two must be matched by size.